### PR TITLE
Include ability to resize, attach, and detach volumes

### PR DESCRIPTION
**Pull request by lalvarezguillen**
See this issue for context: https://github.com/ngine-io/ansible-collection-vultr/issues/15

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -4,6 +4,10 @@ Vultr Collection Release Notes
 
 .. contents:: Topics
 
+v1.1.0
+======
+
+- vultr_block_storage - Included ability to resize, attach and detach volumes.
```
Comment (resmo) [inline CHANGELOG.rst:10]: this is a generated file do not touch. 

please add a changelog fragment to _changelogs/fragments_

e.g. 
```yaml
minor_changes:
  - vultr_block_storage - Included ability to resize, attach and detach volumes.
```
also see https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment

File: plugins/modules/vultr_block_storage.py [hunk 2]
```diff
@@ -37,8 +38,20 @@
...
+  attached_to_SUBID:
+    description:
+      - The ID of the server the volume is attached to.
+      - Required if I(state) is attached.
+    aliases: [ attached_to_id ]
+    type: int
+  live_attachment:
+    description:
+      - Whether the volume should be attached/detached, even if the server not stopped.
+    type: str
```
Comment (resmo) [inline plugins/modules/vultr_block_storage.py:52]: please change this to `bool` (and change the default and choices to be booleans)

File: plugins/modules/vultr_block_storage.py [hunk 3]
```diff
@@ -216,28 +242,140 @@ def absent_block_storage_volume(self):
...
+        volume = self.present_block_storage_volume()
+        if volume.get('attached_to_SUBID') is None:
+            return volume
+
+        self.result['changed'] = True
+
+        if not self.module.check_mode:
+            data = {
+                'SUBID': volume['SUBID'],
+                'live': self.module.params.get('live_attachment')
```
Comment (resmo) [inline plugins/modules/vultr_block_storage.py:255]: as this should be a string of `yes` or `no` (thanks vultr...) and `live_attachment` is bool, you can use the helper method https://github.com/ngine-io/ansible-collection-vultr/blob/master/plugins/module_utils/vultr.py#L125

```
'live': self.get_yes_or_no('live_attachment')
```

File: plugins/modules/vultr_block_storage.py [hunk 4]
```diff
@@ -216,28 +242,140 @@ def absent_block_storage_volume(self):
...
+        if not self.module.check_mode:
+            data = {
+                'SUBID': volume['SUBID'],
+                # This API call expects a param called attach_to_SUBID,
+                # but all the BlockStorage API response payloads call
+                # this parameter attached_to_SUBID. So we'll standardize
+                # to the latter and attached_to_id, but we'll pass the
+                # expected attach_to_SUBID to this API call.
+                'attach_to_SUBID': expected_server,
+                'live': self.module.params.get('live_attachment'),
```
Comment (resmo) [inline plugins/modules/vultr_block_storage.py:294]:
```
'live': self.get_yes_or_no('live_attachment')
```

File: plugins/modules/vultr_block_storage.py [hunk 5]
```diff
@@ -216,28 +242,140 @@ def absent_block_storage_volume(self):
...
         size=dict(type='int'),
         region=dict(type='str'),
-        state=dict(type='str', choices=['present', 'absent'], default='present'),
+        state=dict(
+            type='str',
+            choices=['present', 'absent', 'attached', 'detached'],
+            default='present'
+        ),
+        attached_to_SUBID=dict(type='int', aliases=['attached_to_id']),
+        live_attachment=dict(type='str', choices=['yes', 'no'], default='yes')
```
Comment (resmo) [inline plugins/modules/vultr_block_storage.py:347]:
```
live_attachment=dict(type='bool', default=True)
```

File: plugins/modules/vultr_block_storage.py [hunk 6]
```diff
@@ -216,28 +242,140 @@ def absent_block_storage_volume(self):
...
+
+    def attached_block_storage_volume(self):
+        expected_server = self.module.params.get('attached_to_SUBID')
+        volume = self.present_block_storage_volume()
+        server = volume.get('attached_to_SUBID')
+        if server == expected_server:
+            return volume
+
+        if server is not None:
+            raise VultrException(
```
Comment (resmo) [inline plugins/modules/vultr_block_storage.py:279]: throwing exceptions to users is considered to be a bug in ansible terms. We need to catch it and use 
```
self.module.fail_json(msg="")
```
to send error messages to the user.

**Review by resmo — CHANGES_REQUESTED** [review]
Nice work, a made a few improvements requests. let me know if you want to discuss anything

File: plugins/modules/vultr_block_storage.py (hunk 2 above)
Comment (lalvarezguillen) [reply plugins/modules/vultr_block_storage.py]: fixed

File: plugins/modules/vultr_block_storage.py (hunk 3 above)
Comment (lalvarezguillen) [reply plugins/modules/vultr_block_storage.py]: fixed

File: plugins/modules/vultr_block_storage.py (hunk 4 above)
Comment (lalvarezguillen) [reply plugins/modules/vultr_block_storage.py]: fixed

File: plugins/modules/vultr_block_storage.py (hunk 5 above)
Comment (lalvarezguillen) [reply plugins/modules/vultr_block_storage.py]: fixed

File: plugins/modules/vultr_block_storage.py (hunk 6 above)
Comment (lalvarezguillen) [reply plugins/modules/vultr_block_storage.py]: fixed, and adapted tests to account for this

File: CHANGELOG.rst (hunk 1 above)
Comment (lalvarezguillen) [reply CHANGELOG.rst]: Thanks. I was not familiar with this at all.

I've created the fragment. Please make sure it makes sense to you

**Comment by lalvarezguillen** [comment]
> Nice work, a made a few improvements requests. let me know if you want to discuss anything

Brilliant! Thanks a lot for the review.

I've tackled your observations. Please take another look when you get a chance

**Review by resmo — APPROVED** [review]

**Pull request merged**